### ISSUE TITLE
OpenWeather version not working on RPi3 (but smoothly on RPi4)

### ISSUE THREAD

**Issue by RobbieIoT** [vlaine/Weather-Forecast-Dashboard#7]
Hi,

Long time ago since I replied here, but here I am again!

Last week I used the OpenWeather port (away from Darksky due to ending on the 31st), which runs fine on an RPi4, from where I finetuned it, but when trying to run it on an older RPi3, it doesn't work.

The older Darksky version ran without any issues on the Pi3 since the beginning, but somehow the OpenWeather version not..
I ran update and full-upgrade already, no difference.

What else can I check?

Thank you in advance!

**Comment by RobbieIoT** [comment]
As an addition; it only shows only 'hum %' and nothing else:

![Weather-Forecast-V2](https://user-images.githubusercontent.com/36163166/229496852-b6808fb1-14c7-4078-b3b2-11f83a051d6b.png)

**Comment by vlaine** [comment]
On the one that is not working, are you using a new OpenWeatherMap account ?  I discovered that the default api key generated when creating an account never activates.  You have to create a new one, delete the default one and wait for 2 hours before it activates.

**Comment by RobbieIoT** [comment]
Hi,

The account is valid, as said, on the Pi4's it's running like on Darksky;

![Weather-Forecast-V2-2](https://user-images.githubusercontent.com/36163166/229497668-e62ac617-87b4-4800-8744-0d0fd52a49d1.png)
![Weather-Forecast-V2-3](https://user-images.githubusercontent.com/36163166/229497688-c81dd3f2-dc2e-44bb-8f19-eaa42ab6fd37.png)

I have this account for a couple of years now (even before you ported the Darksky to OpenWaether)...

**Comment by RobbieIoT** [comment]
Two things I also tried:
- Update to use newer googleapi lib jquery.min.js 3.6.3
- Installed Firefox to check it, same result

Testing now with the current version, without anything altered (first I thought that might have been the cause...)

**Comment by vlaine** [comment]
From the test I performed, free api keys that were active before onecall was migrated to the version 3.0 are still working for free with the version 2.5.  However, new api keys generated are not supporting onecall and this is really annoying because the forecast is not giving half of the information I am displaying in the dashboard.  I have to think about this.

The free calls included with onecall 3.0 would be enough for one weather station, but I would like to make sure they offer spending limits before subscribing to prevent any unwanted charges.

**Comment by RobbieIoT** [comment]
Hi vlaine,

Thank you for putting time into this!

I hope you don't miss one important remark I made; on the Pi4 it's working perfectly, so in my opinion it doesn't have to do anything with the key I use...

It must be caused by a difference between the Pi3 and Pi4 (OS)? 

I saw you also made several updates when porting to OpenWeather; might one of these changes have to do with it?

But I'm just guessing though...

**Comment by vlaine** [comment]
Are you using the same api key for both device?  If yes, you are right, the problem is not related to the key and we should look elsewhere.  The HTML and JavaScript are very basic and should work on a Pi3 but everything is possible at this point.  Double check that your pi3 key can make a onecall (v2.5) call.  I was able to reproduce the same behaviour using an api key recently created and I just want to confirm it’s not subscription related.

**Comment by BadArms37** [comment]
Hi ElsecoBeheer!

Are there any error messages in the browser's dev tools console?

**Comment by vlaine** [comment]
Debugging using Chromium on a pi3 can be a journey. A easy test would be to try this url in your pi 3 browser.  
http://api.openweathermap.org/data/2.5/onecall?lat=40.730610&lon=-73.935242&units=metric=&APPID=yourApiKey

If you get an error 401, the api key does not support onecall.  If you get a json response, paste the result here

**Comment by RobbieIoT** [comment]
@BadArms37 yes, there are. But these are there already from the first start couple of years ago, with DarkSky.

@vlaine yes, same key, that's why I pointed it out again.

And did that also; copied link and tested it, was one of the first things I tried. Also played with changing the Onecall option, all worked and just resulted in a differend kind of results. No error messages...

Also, when requesting the weather with the DarkSky version, I get to see the layout (like av. temp, windspeed etc. as in the 2nd/3rd screenshot), but without values and no hourly columns...
When trying the new OpenWeather version, I only get to see like the first screenshot (hum %). So, it's incomplete and without the translated words.
I never used the humidity option, so it is strange it's shown. I got the feeling something is breaking in that corner on the Pi3 and because of that things are missing (exiting script ?)

**Comment by vlaine** [comment]
Onecall 2.5 is phasing out anyway so I have to port my code to onecall 3.0.  I received the confirmation that you can set the limit to 1000 calls per day to prevent charges so it will remain free.  I will let you know when the code is updated and hopefully it will fix your problem.

**Comment by RobbieIoT** [comment]
Hi vlaine,

Can I assist in any way?

Br. Robert

**Comment by vlaine** [comment]
Issue closed by mistake and reopened immediately. Sorry about that.

Prepare your account.  Go to the billing section.  Set the onecall 3.0 limit to 1000.  Wait 24h and add your billing information.  
My instinct is that it’s a simple url change in my code.  If you want to test it before me, go into the script.js and change the url (change 2.5 for 3.0).  
on your pi 3, I suggest a full reinstall just in case something was not properly configured.

**Comment by vlaine** [comment]
The code has been updated to onecall 3.0 and api errors are now displayed in the alert section to facilitate debugging.  I hope it will help you.

**Comment by BadArms37** [comment]
Installed the latest version on my R Pi 4 / latest RaspOS 64-bit and it worked fine.  Note that there were no errors displayed for me in Chromium's dev tool console.  

I did observe the "hum %" thing when I attempted to launch the index.html from a link on my desktop.  My link's location was (incorrectly) set to the desktop's location which resulted in several "ERR_FILE_NOT_FOUND" errors showing in the console.

I should have access to a RPi 3 soon and will test that one shortly.

**Comment by RobbieIoT** [comment]
Hi vlaine,

Not able to test it before tomorrow, need credit card information to start the '1000 free calls a day' and not at the office today.
In the mean time, can you describe a littlebit what you updated, so I can learn from it?

Thank you in advance!

**Comment by RobbieIoT** [comment]
> Installed the latest version on my R Pi 4 / latest RaspOS 64-bit and it worked fine. Note that there were no errors displayed for me in Chromium's dev tool console.
> 
> I did observe the "hum %" thing when I attempted to launch the index.html from a link on my desktop. My link's location was (incorrectly) set to the desktop's location which resulted in several "ERR_FILE_NOT_FOUND" errors showing in the console.
> 
> I should have access to a RPi 3 soon and will test that one shortly.

Hi,

Yes, same for me. On a Pi4 it runs smoothly (as you can see in some screenshots above). Did not test the OneCall 3.0 version yet, but hopefully able tomorrow after adding credit card info.

Very curious which result you get with a Pi3 (especially the One Call 2.5 which I'm running now)...

**Comment by vlaine** [comment]
I changed the api url to use onecall 3.0 because version 2.5 is explicitly blocked for new users / api keys.  I added code to display the api error message in the red alert bar to help debugging.  The remaining of the code is unchanged.

**Comment by BadArms37** [comment]
Tested latest code on R Pi 3B (Raspberry OS 32 bit + Chromium).  No issues / worked great.
Also tested on an older Intel x86 running CentOS Stream 8 + Chromium.  No issues / worked great.

**Comment by RobbieIoT** [comment]
Hi vlaine,

It took a long time, but the issue I experienced was yesterday discovered by a colleague developer;

![Afbeelding van WhatsApp op 2023-08-15 om 16 17 44](https://github.com/vlaine/Weather-Forecast-Dashboard/assets/36163166/259576cc-401b-4e55-b5e8-e27a37b3874c)

The ?? function you used is only supported starting from chromium version 80, but the RPi3 is still stuck on version 72.
This was the reason the information was not shown properly.

We rewrote the script a littlebit to make it functional for the older chromium version.

It's all running fine now, so I wanted to share this info with you!

Thank you for all your support!
